### ISSUE TITLE
The training script and the description in the paper are different.

### ISSUE THREAD

**Issue by MaverickRen** [dvlab-research/LISA#41]
Nice work! I notice that you present two versions of results in Table 1 of the main paper. One version is called (ft) and described as training without any ReasonSeg data first and then fine-tuned only on 239 ReasonSeg data. But i notice that the training script in this repo use ReasonSeg by default. The result I got using this script is between the (ft) version and the non-(ft) version. Is this result inconsistent with the paper?

**Comment by X-Lai** [comment]
Yes, they are inconsistent. Currently, the released `v1` model is our latest model. If you want to reproduce the results, you may try to use the `v0` model. We will also update the results in the paper soon.

**Comment by MaverickRen** [comment]
Is v0 the version before the code rewrite on August 23rd? I actually use v0, so if I want to reproduce the non-ft results in the paper, I have to remove the ReasonSeg data set from the training set, right?

**Comment by MaverickRen** [comment]
To reproduce the results of the ft version in the paper, how should I configure the fine-tuning on ReasonSeg, including the number of training steps and learning rate?

**Comment by X-Lai** [comment]
> Is v0 the version before the code rewrite on August 23rd? I actually use v0, so if I want to reproduce the non-ft results in the paper, I have to remove the ReasonSeg data set from the training set, right?

Yes, you're correct.

**Comment by X-Lai** [comment]
> To reproduce the results of the ft version in the paper, how should I configure the fine-tuning on ReasonSeg, including the number of training steps and learning rate?

Other configurations remain the same.

**Comment by SxJyJay** [comment]
> Nice work! I notice that you present two versions of results in Table 1 of the main paper. One version is called (ft) and described as training without any ReasonSeg data first and then fine-tuned only on 239 ReasonSeg data. But i notice that the training script in this repo use ReasonSeg by default. The result I got using this script is between the (ft) version and the non-(ft) version. Is this result inconsistent with the paper?

I get the same experimental results with you. I have tried to reproduce both LISA-7B and LISA-llama2-13B, and the results are 7B: `giou=0.502, ciou=0.533`; llama2-13B: `giou=0.540, ciou=0.608`. Both are lower than the reported ones. Have you @MaverickRen found any solution? Thanks!

**Comment by MaverickRen** [comment]
> > Nice work! I notice that you present two versions of results in Table 1 of the main paper. One version is called (ft) and described as training without any ReasonSeg data first and then fine-tuned only on 239 ReasonSeg data. But i notice that the training script in this repo use ReasonSeg by default. The result I got using this script is between the (ft) version and the non-(ft) version. Is this result inconsistent with the paper?
> 
> I get the same experimental results with you. I have tried to reproduce both LISA-7B and LISA-llama2-13B, and the results are 7B: `giou=0.502, ciou=0.533`; llama2-13B: `giou=0.540, ciou=0.608`. Both are lower than the reported ones. Have you @MaverickRen found any solution? Thanks!

I just reproduce the LISA-7B version using the script given in this repo, and the results is giou=0.508, ciou=0.516. According to the author's previous reply, i think this is reasonable. The ft-version probably contains two stages of training. The first stage does not use reasonseg, and the second stage only uses reasonseg. I

**Comment by SxJyJay** [comment]
Got it. Thanks a lot!

**Comment by ryusaeba** [comment]
Hi @MaverickRen, 

Did you try to use two stage training processs to get higher accuracy already? If so, is the following just what you did?
Stage1: 
--dataset="sem_seg||refer_seg||vqa||reason_seg" with "sample_rates=9,3,3,0"
or
--dataset="sem_seg||refer_seg||vqa" with "sample_rates=9,3,3"

Stage2: 
--dataset="sem_seg||refer_seg||vqa||reason_seg" with "sample_rates=0,0,0,1"
or
--dataset="reason_seg" with "sample_rates=1"

**Comment by wyn-Mandy** [comment]
> Yes, they are inconsistent. Currently, the released `v1` model is our latest model. If you want to reproduce the results reported in our paper, you may try to use the `v0` model. We will also update the results in the paper soon.是的，它们不一致。目前发布的 `v1` 模型是我们最新的模型。如果你想要复现论文中报告的结果，你可以尝试使用 `v0` 模型。我们也会尽快更新论文中的结果。

你好，我正在尝试复现lisa。目前我在两张4090上搭建环境，torch                     2.8.0，torchvision 0.23.0，我的训练命令是：
deepspeed --master_port=24999 train_ds.py \
  --version="./LLaVA-Lightning-7B-delta-v1-1" \
  --dataset_dir='./dataset' \
  --vision_pretrained="PATH_TO_SAM" \
  --dataset="sem_seg||refer_seg||vqa" \
  --sample_rates="9,3,3" \
  --exp_name="lisa-7b"，
batch等参数保持默认，10 epoch的iou一直是0.002和0.000，请问这是为什么？我也尝试了在一张4090严格按照requirements.txt配置环境，训练得到的mask_loss仍然是震荡的 并非稳定下降。
期待您的回复！

**Comment by wyn-Mandy** [comment]
> > > Nice work! I notice that you present two versions of results in Table 1 of the main paper. One version is called (ft) and described as training without any ReasonSeg data first and then fine-tuned only on 239 ReasonSeg data. But i notice that the training script in this repo use ReasonSeg by default. The result I got using this script is between the (ft) version and the non-(ft) version. Is this result inconsistent with the paper?干得漂亮！我注意到你在主论文的表 1 中展示了两个版本的结果。其中一个版本称为（ft），描述为首先不使用任何 ReasonSeg 数据进行训练，然后仅使用 239 个 ReasonSeg 数据进行微调。但我注意到这个仓库中的训练脚本默认使用 ReasonSeg。我使用这个脚本得到的结果介于（ft）版本和非（ft）版本之间。这个结果与论文不一致吗？
> > 
> > 
> > I get the same experimental results with you. I have tried to reproduce both LISA-7B and LISA-llama2-13B, and the results are 7B: `giou=0.502, ciou=0.533`; llama2-13B: `giou=0.540, ciou=0.608`. Both are lower than the reported ones. Have you [@MaverickRen](https://github.com/MaverickRen) found any solution? Thanks!我和你得到了相同的实验结果。我尝试复现了 LISA-7B 和 LISA-llama2-13B，结果分别是 7B: `giou=0.502, ciou=0.533` ; llama2-13B: `giou=0.540, ciou=0.608` 。这两个结果都低于报告的数值。你找到解决方案了吗？谢谢！
> 
> I just reproduce the LISA-7B version using the script given in this repo, and the results is giou=0.508, ciou=0.516. According to the author's previous reply, i think this is reasonable. The ft-version probably contains two stages of training. The first stage does not use reasonseg, and the second stage only uses reasonseg. I我仅使用这个仓库中提供的脚本重新生成了 LISA-7B 版本，结果为 giou=0.508，ciou=0.516。根据作者之前的回复，我认为这是合理的。ft 版本可能包含两个阶段的训练。第一阶段不使用 reasonseg，第二阶段仅使用 reasonseg。

你好，方便问一下您复现LISA-7B的显卡和torch版本吗，我按照readme的说明下载了全部数据集，但是训练结果非常不理想，mask_loss一直是震荡的。
